compute/reservations/consume_any_reservation.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_consume_any_matching_reservation]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	"google.golang.org/protobuf/proto"
+)
+
+// Creates the reservation from given template in particular zone
+func createConsumableReservation(w io.Writer, projectID, zone, reservationName, sourceTemplate string) error {
```

Comment:
issue: it looks like you're duplicating the same code, `createConsumableReservation()` twice. Let's not do that. If the snippet needs this to run, then put all of this code into the test file.

Reply:
I've done it in this way, to be as close as possible to the [gcloud example in documentation](https://cloud.google.com/compute/docs/instances/reservations-consume#consuming_a_specific_single-project_reservation). 
There it is done in 2 steps:
1) shows how to create a reservation (yeah, in this case, consuming any reservation, it just duplicates existing sample)
2) actual consuming

Therefore, I thought it would be consistent to keep 2 functions. Do you think it's better to remove first and keep only **consumeAnyReservation()**?

Reply:
Yes. The major issue here is maintenance. 

Consider this hypothetical: we have >1 samples that demonstrate how to create a reservation. Then we get a breaking change to the underlying API. That's more than 1 potential point of failure that we need to resolve.

If the docs want to show both creating and consuming a reservation, then they can import each atomic sample individually.

Reply:
I see your point. 
Adjusted samples and instead of additional function, left a note in comment to highlight a requirement for reservation, which is expected to be consumed.